src/components/Visualization/index.js

```diff
@@ -198,16 +197,18 @@ class VisualizationView extends React.Component {
                         // which corresponds to a row of data visualized.
                         listeners[event] = (d) => {
 
-                            let graphQueryParams = {};
-                            let resetFilters = false;
+                            const GraphComponent = GraphManager.getGraphComponent(configuration.graph)
 
-                            if(configuration.key) {
-                                let vizID = `${id.replace(/-/g, '')}vkey`;
-                                let vKey = evalExpression("(" + configuration.key + ")")(d);
-                                if(this.props.orgContext[vizID] === vKey)
-                                    resetFilters = true;
+                            let graphQueryParams = {},
+                                resetFilters = false,
+                                vizID = `${id.replace(/-/g, '')}vkey`,
+                                vKey = GraphComponent.getGraphKey(configuration);
 
-                                graphQueryParams[vizID] = vKey;
+                            if(vKey) {
+                                if(this.props.orgContext[vizID] === vKey(d)) {
```

Comment:
instead of calling vKey(d) twice let's cache the result and use in both case: if statement and below.